### PR TITLE
[SDK-980] fetch bodytype specific templates

### PR DESCRIPTION
**Pull request by rk132**
## Description

-   Fetch body type specific templates

## How to Test

-   Verify, that correct templates are selected

[SDK-980]: https://ready-player-me.atlassian.net/browse/SDK-980?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Runtime/AvatarCreator/Scripts/WebRequests/AvatarAPIRequests.cs [hunk 1]
```diff
@@ -66,7 +63,7 @@ public async Task<List<AvatarTemplateData>> GetAvatarTemplates()
             var response = await authorizedRequest.SendRequest<Response>(
                 new RequestData
                 {
-                    Url = $"{RPM_AVATAR_V2_BASE_URL}/templates",
+                    Url = $"{RPM_AVATAR_V2_BASE_URL}/templates?{BODY_TYPE}={CoreSettingsHandler.CoreSettings.BodyType.GetDescription()}",
```
Comment (HarrisonHough) [inline Runtime/AvatarCreator/Scripts/WebRequests/AvatarAPIRequests.cs:66]: is it possible that we might use this body type parameter somewhere else? 
If so we could just make a function on Core settings like 
CoreSettings.GetBodyTypeParameter 
that would return this 
$"bodyType={bodyType.GetDescription}"

File: Runtime/AvatarCreator/Scripts/WebRequests/AvatarAPIRequests.cs [hunk 2]
```diff
@@ -83,7 +80,7 @@ public async Task<AvatarProperties> CreateFromTemplateAvatar(string templateId,
             var payloadData = new Dictionary<string, string>
             {
                 { nameof(partner), partner },
-                { nameof(bodyType), GetBodyTypeValue(bodyType) }
+                { nameof(bodyType), bodyType.GetDescription() }
```
Comment (HarrisonHough) [inline Runtime/AvatarCreator/Scripts/WebRequests/AvatarAPIRequests.cs:83]: does this CreateFromTemplateAvatar function really need a BodyType parameter now? shouldn't it be coming from core settings? 🤔

File: Runtime/Core/Scripts/Extensions/EnumExtensions.cs [hunk 3]
```diff
@@ -24,5 +25,15 @@ public static class EnumExtensions
 
             return default;
         }
+
+        public static string GetDescription(this Enum value)
```
Comment (HarrisonHough) [inline Runtime/Core/Scripts/Extensions/EnumExtensions.cs:29]: we actually already have this function in the BodyShapeSelectionElement.cs class. However I think that a better location for it is here as you have it. So can you remove the other one from BodyShapeSelection and make it use this one? 
Screenshot 2024-06-18 at 8 30 58 am

**Review by HarrisonHough — CHANGES_REQUESTED** [review]
looks good overall. The only thing is that we already have a GetEnumDescription function, see my comments on that. 
I also had some other questions in the comments.

File: Runtime/AvatarCreator/Scripts/WebRequests/AvatarAPIRequests.cs (hunk 1 above)
Comment (rk132) [reply Runtime/AvatarCreator/Scripts/WebRequests/AvatarAPIRequests.cs:66]: We are still storing the avatar bodyType within the core in some places. So I don't think that's a good way to go about it for now.

Comment (HarrisonHough) [reply Runtime/AvatarCreator/Scripts/WebRequests/AvatarAPIRequests.cs:66]: okay makes sense lets leave as it is then.

**Review by HarrisonHough — APPROVED** [review]
Looks good, nice work!

**Pull request merged**